libs-scala/ports/src/main/scala/com/digitalasset/ports/PortFiles.scala

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ports
+
+import java.nio.file.{Files, Path}
+
+import scalaz.{Show, \/}
+
+import scala.collection.JavaConverters._
+
+object PortFiles {
+  sealed abstract class Error extends Serializable with Product
+  final case class FileAlreadyExists(path: Path) extends Error
+  final case class CannotWriteIntoFile(path: Path, reason: String) extends Error
```

Comment:
I think `CannotWriteToFile` is slightly more idiomatic, but not fussed.

Reply:
Tell me what you want to cal it... I wanted to have `FileAlreadyExists` as a separate error, don't care much about the other error.

Reply:
Eh, if the build passes, merge; if it fails, push the change. 😛